xarray/core/npcompat.py

```diff
@@ -1,260 +1,8 @@
-
-        Returns
-        -------
-        nancumprod : ndarray
-            A new array holding the result is returned unless `out` is
-            specified, in which case it is returned.
-
-        See Also
-        --------
-        numpy.cumprod : Cumulative product across array propagating NaNs.
-        isnan : Show which elements are NaN.
-
-        Examples
-        --------
-        >>> np.nancumprod(1)
-        array([1])
-        >>> np.nancumprod([1])
-        array([1])
-        >>> np.nancumprod([1, np.nan])
-        array([ 1.,  1.])
-        >>> a = np.array([[1, 2], [3, np.nan]])
-        >>> np.nancumprod(a)
-        array([ 1.,  2.,  6.,  6.])
-        >>> np.nancumprod(a, axis=0)
-        array([[ 1.,  2.],
-               [ 3.,  2.]])
-        >>> np.nancumprod(a, axis=1)
-        array([[ 1.,  2.],
-               [ 3.,  3.]])
-
-        """
-        a, mask = _replace_nan(a, 1)
-        return np.cumprod(a, axis=axis, dtype=dtype, out=out)
-
-    def flip(m, axis):
-        """
-        Reverse the order of elements in an array along the given axis.
-
-        The shape of the array is preserved, but the elements are reordered.
-
-        .. versionadded:: 1.12.0
-
-        Parameters
-        ----------
-        m : array_like
-            Input array.
-        axis : integer
-            Axis in array, which entries are reversed.
-
-
-        Returns
-        -------
-        out : array_like
-            A view of `m` with the entries of axis reversed.  Since a view is
-            returned, this operation is done in constant time.
-
-        See Also
-        --------
-        flipud : Flip an array vertically (axis=0).
-        fliplr : Flip an array horizontally (axis=1).
-
-        Notes
-        -----
-        flip(m, 0) is equivalent to flipud(m).
-        flip(m, 1) is equivalent to fliplr(m).
-        flip(m, n) corresponds to ``m[...,::-1,...]`` with ``::-1`` at index n.
-
-        Examples
-        --------
-        >>> A = np.arange(8).reshape((2,2,2))
-        >>> A
-        array([[[0, 1],
-                [2, 3]],
-
-               [[4, 5],
-                [6, 7]]])
-
-        >>> flip(A, 0)
-        array([[[4, 5],
-                [6, 7]],
-
-               [[0, 1],
-                [2, 3]]])
-
-        >>> flip(A, 1)
-        array([[[2, 3],
-                [0, 1]],
-
-               [[6, 7],
-                [4, 5]]])
-
-        >>> A = np.random.randn(3,4,5)
-        >>> np.all(flip(A,2) == A[:,:,::-1,...])
-        True
-        """
-        if not hasattr(m, 'ndim'):
-            m = np.asarray(m)
-        indexer = [slice(None)] * m.ndim
-        try:
-            indexer[axis] = slice(None, None, -1)
-        except IndexError:
-            raise ValueError("axis=%i is invalid for the %i-dimensional "
-                             "input array" % (axis, m.ndim))
-        return m[tuple(indexer)]
+as_strided = np.lib.stride_tricks.as_strided
```

Comment:
We could push this down into the caller, rather than maintain a layer here (but also fine as-is!)